extends Area2D

@onready var player: CharacterBody2D = get_tree().get_first_node_in_group("Player")
@onready var tilemap = $"../TileMap"

var is_open: bool = false

@export var tiles_to_remove: Array[Vector2i] = [
	Vector2i(-4, 0),
	Vector2i(-4, 1),
	Vector2i(-4, 2),
	Vector2i(-4, 3)
]

func _on_body_entered(body):
	if body.is_in_group("Player") and not is_open:
		print("C·ª≠a m·ªü th√†nh c√¥ng üéâ")
		open_door()

func open_door():
	is_open = true
	# X√≥a tile c·ª≠a
	for coords in tiles_to_remove:
		tilemap.set_cell(0, coords, -1)

	# Sau 1 gi√¢y c·ª≠a ƒë√≥ng l·∫°i
	await get_tree().create_timer(2).timeout
	close_door()

func close_door():
	is_open = false
	# ƒê·∫∑t l·∫°i tile c·ª≠a b·∫±ng ID ƒë√£ hardcode s·∫µn
	for coords in tiles_to_remove:
		tilemap.set_cell(0, coords, 0, Vector2i(2, 9))
